block_issuer_key.go

```diff
@@ -0,0 +1,51 @@
+package iotago
+
+import (
+	"bytes"
+	"sort"
+
+	"github.com/iotaledger/hive.go/crypto/ed25519"
+	"github.com/iotaledger/hive.go/serializer/v2"
+)
+
+// BlockIssuerKeyType defines the type of block issuer key.
+type BlockIssuerKeyType byte
+
+const (
+	// Ed25519BlockIssuerKey denotes a BlockIssuerKeyEd25519.
+	Ed25519BlockIssuerKey BlockIssuerKeyType = iota
+)
+
+// BlockIssuerKeys are the keys allowed to issue blocks from an account with a BlockIssuerFeature.
+type BlockIssuerKeys []BlockIssuerKey
+
+func (keys BlockIssuerKeys) Sort() {
+	sort.Slice(keys, func(i, j int) bool {
+		return bytes.Compare(keys[i].PublicKeyBytes(), keys[j].PublicKeyBytes()) < 0
```

Comment:
Should we maybe compare the Bytes instead of the PublicKeyBytes? I mean the type byte added as well.
What if there are two different types, but they use the same bytes for the pubkey? Then sorting is not guranteed.